CMPUT301Project/app/src/main/java/com/example/GroupProject/ProfileFragment.java

```diff
@@ -31,24 +60,96 @@ public ProfileFragment() {
      * @return A new instance of fragment ProfileFragment.
      */
     // TODO: Rename and change types and number of parameters
-    public static ProfileFragment newInstance(String param1, String param2) {
+    public static ProfileFragment newInstance(String username) {
         ProfileFragment fragment = new ProfileFragment();
         Bundle args = new Bundle();
+        args.putString("username", username);
         fragment.setArguments(args);
         return fragment;
     }
 
+    @Override
+    public void onAttach(@NonNull Context context) {
+        super.onAttach(context);
+        thisContext = context;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        FirebaseFirestore db;
+        db = getFirestoreInstance();
+
+        confirmUserbtn.setOnClickListener(view1 -> {
+            Map<String, Object> user = new HashMap<>();
+            if (usernameEdit.getText() != null) {
+                String newUsername = usernameEdit.getText().toString();
+                thisUser.setUsername(newUsername);
+            }
+            if (fnameEdit.getText() != null) {
+                String newFname = fnameEdit.getText().toString();
+                thisUser.setFirstName(newFname);
+            }
+            if (lnameEdit.getText() != null) {
+                String newLname = lnameEdit.getText().toString();
+                thisUser.setLastName(newLname);
+            }
+            user.put("FirstName", thisUser.getFirstName());
```

Comment:
What happens when one of the text fields is null and the property is not set? Then there is no value to get.

Reply:
You might want to disable the button unless all text fields have a value.